Library broken with Angular v10: `ModuleWithProviders<T>` requires type argument

### Reproduction link
[https://github.com/angular/ngcc-validation/pull/1241](https://github.com/angular/ngcc-validation/pull/1241)

### Steps to reproduce
With the project from [ngcc-validation/projects/ng-zorro-antd-ngcc][1] and the dependency versions shown in [ngcc-validation/package.json][2] (note we are mainly interested in the versions of `@angular/*`, `ng-zorro-antd` and TypeScript, run:

```
yarn install
yarn ng e2e ng-zorro-antd-ngcc
```

[1]: https://github.com/angular/ngcc-validation/tree/2759ae3c8c03cc304fad3b3316e48bff9c36e70c/projects/ng-zorro-antd-ngcc
[2]: https://github.com/angular/ngcc-validation/blob/2759ae3c8c03cc304fad3b3316e48bff9c36e70c/package.json

### What is expected?
Building the app should succeed.

### What is actually happening?
Building the app fails.

| Environment | Info |
|---|---|
| ng-zorro-antd | 9.1.2 |
| Browser | Any |

---
With PR https://github.com/angular/angular/pull/36963, the type argument was made mandatory for `ModuleWithProviders`. (This was documented as a breaking change in https://github.com/angular/angular/pull/37121.)

In certain cases, ngcc (the Angular Compatibility Compiler) is able to infer the type parameter, but not when used for an arbitrary type. [component-bed.ts][1] uses `ModuleWithProviders` without a type argument as part of the `ComponentDeps` type, which causes a build error when building against Angular v10.

To avoid the build error, you must pass a type argument to `ModuleWithProviders` (e.g. `ModuleWithProviders<any>` or `ModuleWithProviders<unknown>`).

##
For reference, we run into this issue on our [ngcc-validation][2] repo. Here is an example of a failing CI job: https://circleci.com/gh/angular/ngcc-validation/8853#tests/containers/3

```
ERROR in node_modules/ng-zorro-antd/core/testing/componet-bed.d.ts:4:54 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).

4 declare type ComponentDeps = Array<Type<NzSafeAny> | ModuleWithProviders>;
                                                       ~~~~~~~~~~~~~~~~~~~
```

[1]: https://github.com/NG-ZORRO/ng-zorro-antd/blob/f99d7ff3578e05d1382e9a36ebd805a255383bac/components/core/testing/componet-bed.ts#L7
[2]: https://github.com/angular/ngcc-validation
<!-- generated by ng-zorro-issue-helper. DO NOT REMOVE -->

--- a/components/core/testing/componet-bed.ts
+++ b/components/core/testing/componet-bed.ts
@@ -1,10 +1,9 @@
 import { CommonModule } from '@angular/common';
-import { DebugElement, ModuleWithProviders, NO_ERRORS_SCHEMA, Provider, Type } from '@angular/core';
+import { DebugElement, NgModule, NO_ERRORS_SCHEMA, Type } from '@angular/core';
 import { ComponentFixture, TestBed, TestBedStatic } from '@angular/core/testing';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
-import { NzSafeAny } from 'ng-zorro-antd/core/types';
 
-type ComponentDeps = Array<Type<NzSafeAny> | ModuleWithProviders>;
+type ComponentBedOptions = Pick<NgModule, 'providers' | 'declarations' | 'imports'>;
 export interface ComponentBed<T> {
   bed: TestBedStatic;
   fixture: ComponentFixture<T>;
@@ -14,11 +13,7 @@ export interface ComponentBed<T> {
 }
 export function createComponentBed<T>(
   component: Type<T>,
-  options: {
-    providers?: Provider[];
-    declarations?: ComponentDeps;
-    imports?: ComponentDeps;
-  } = {
+  options: ComponentBedOptions = {
     providers: [],
     declarations: [],
     imports: []

